### DISCUSSION TITLE
Authorizing Outbound Traffic Imperatively

### DISCUSSION THREAD

**Discussion by ralphotowo** [BretFisher/ama#205] [Q&A]
Thanks to your course, I've deployed a 2-node Kubernetes cluster to Digital Ocean with my Node web application and successfully exposed the service. I'm able to access my web app but the app is unable to access its database. 

I've searched for solutions but everything I've found recommends applying a network policy with an egress section defining the ports needed. I would like to run a CLI kubectl command to allow outbound connections on TCP:3306, is there a straight forward way to do this without writing YAML?

**Comment by ralphotowo** [comment]
The issue here for me wasn't with Kubernetes itself. I had started a Kuberenetes cluster on Digital Ocean with multiple nodes, this meant that D.O spawned multiple droplets as part of my cluster. The database server was running off an AWS EC2 instance and I needed to white-list the public IP addresses of all the droplets before my application was able to access the database on AWS.

While this is working for now, I'm a little concerned about the need to white-list the IPs on AWS. What happens if I'm using a cloud provider where an auto scaling policy is able to dynamically spawn nodes depending on resource consumption?

**Comment by BretFisher** [comment]
A few thoughts:
1. it's an anti-pattern to have your web frontend and db backend in two different clouds or regions.
2. This is why it's advisable to stick with one cloud provider where these things are usually automated. Once you try to use a cross-cloud approach, things like this get much harder and require much more infrastructure and automation that you build yourself to sync this type of configuration.
